src/commands/mod.rs

```diff
@@ -26,33 +27,76 @@ impl DaemonControl {
 
     /// Get a new deposit address. This will always generate a new deposit address, regardless of
     /// whether it was actually used.
-    pub fn get_new_address(&self) -> bitcoin::Address {
+    pub fn get_new_address(&self) -> GetAddressResult {
         let mut db_conn = self.db.connection();
         let index = db_conn.derivation_index();
         // TODO: handle should we wrap around instead of failing?
         db_conn.update_derivation_index(index.increment().expect("TODO: handle wraparound"));
-        self.config
+        let address = self
+            .config
             .main_descriptor
             // TODO: have a descriptor newtype along with a derived descriptor one.
             .derive(index.into())
             .translate_pk2(|xpk| xpk.derive_public_key(&self.secp))
             .expect("All pubkeys were derived, no wildcard.")
-            .address(self.config.bitcoind_config.network)
-            .expect("It's a wsh() descriptor")
+            .address(self.config.bitcoin_config.network)
+            .expect("It's a wsh() descriptor");
+        GetAddressResult { address }
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize)]
 pub struct GetInfoDescriptors {
     pub main: descriptor::Descriptor<descriptor::DescriptorPublicKey>,
 }
 
 /// Information about the daemon
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize)]
 pub struct GetInfoResult {
     pub version: String,
     pub network: bitcoin::Network,
     pub blockheight: i32,
     pub sync: f64,
     pub descriptors: GetInfoDescriptors,
 }
+
+#[derive(Debug, Clone, Serialize)]
```

Comment:
Please add Deserializer also, It is useful for the gui to reuse the struct

Reply:
Same for GetInfoResult

Reply:
Hmm i'll do it here for what is touched, but for the other results i'll make a new PR.

Reply:
In the end i just added a commit here.